fuzz/src/lib.rs

```diff
@@ -150,13 +142,57 @@ pub trait DbSimulator {
 				}
 			}
 		}
-		if let Err(e) =
-			check_db_and_model_are_equals(&db, Self::model_content(&model), is_db_b_tree).unwrap()
-		{
+		if let Err(e) = Self::check_db_and_model_are_equals(&db, &model, is_db_b_tree).unwrap() {
 			panic!("Not able to recover to a proper state when coming back from an instrumented failure: {}", e)
 		}
 		(db, model)
 	}
+
+	fn check_db_and_model_are_equals(
+		db: &parity_db::Db,
+		model: &Self::Model,
+		is_db_b_tree: bool,
+	) -> parity_db::Result<Result<(), String>> {
+		for (k, v) in Self::model_required_content(model) {
+			if db.get(0, &k)?.as_ref() != Some(&v) {
+				return Ok(Err(format!("The value {:?} for key {:?} is not in the database", k, v)))
+			}
+		}
+
```

Comment:
We should keep reference to removed content in model and explicitely check (when not ref_counted) that the removed content in model is not in db anymore.

Reply:
Done.